deploy

```diff
@@ -132,37 +132,49 @@ def get_username():
 
 
 def process_template(args):
-    params = []
+    params = {}
     if not args.official:
-        params += [f'DEVELOPER_PREFIX={args.prefix}']
+        params['DEVELOPER_PREFIX'] = args.prefix
     if args.pipeline:
-        params += params_from_git_refspec(args.pipeline, 'JENKINS_S2I')
-        params += params_from_git_refspec(args.pipeline, 'JENKINS_JOBS')
+        params.update(params_from_git_refspec(args.pipeline, 'JENKINS_S2I'))
+        params.update(params_from_git_refspec(args.pipeline, 'JENKINS_JOBS'))
     if args.config:
-        params += params_from_git_refspec(args.config, 'FCOS_CONFIG')
+        params.update(params_from_git_refspec(args.config, 'FCOS_CONFIG'))
     if args.bucket:
-        params += [f'S3_BUCKET={args.bucket}']
+        params['S3_BUCKET'] = args.bucket
     if args.cosa_img:
-        params += [f'COREOS_ASSEMBLER_IMAGE={args.cosa_img}']
+        params['COREOS_ASSEMBLER_IMAGE'] = args.cosa_img
     if args.pvc_size:
-        params += [f'PVC_SIZE={args.pvc_size}']
+        params['PVC_SIZE'] = args.pvc_size
     if args.kvm_selector:
-        params += [f'KVM_SELECTOR={args.kvm_selector}']
+        params['KVM_SELECTOR'] = args.kvm_selector
 
     print("Parameters:")
-    for param in params:
-        print(f"  {param}")
-    params = [q for p in params for q in ['--param', p]]
+    for k, v in params.items():
+        print(f"  {k}={v}")
     print()
 
-    resources = subprocess.check_output(
-        ['oc', 'process', '--filename', 'manifests/pipeline.yaml'] + params)
-    return json.loads(resources)
+    def gen_param_args(selected):
+        return [q for k, v in params.items() if k in selected for q in ['--param', f'{k}={v}']]
```

Comment:
I find this hard to follow - maybe splitting into two operations would make things clearer

Reply:
Done!